crates/gossip/src/timeout.rs

```diff
@@ -0,0 +1,155 @@
+use crate::gossip::K2Gossip;
+use crate::state::GossipRoundState;
+use crate::K2GossipConfig;
+use kitsune2_api::Url;
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::sync::Mutex;
+use tokio::task::AbortHandle;
+
+/// Spawns a task that checks for timed-out gossip rounds and drops their state.
+pub(crate) fn spawn_timeout_task(
+    config: Arc<K2GossipConfig>,
+    gossip: K2Gossip,
+) -> AbortHandle {
+    tracing::info!("Starting timeout task");
+
+    let round_timeout = config.round_timeout();
+    tokio::spawn(async move {
+        loop {
+            // Check for timed out rounds every 5s
+            tokio::time::sleep(Duration::from_secs(5)).await;
+
+            remove_timed_out_rounds(
+                round_timeout,
+                gossip.initiated_round_state.clone(),
+                gossip.accepted_round_states.clone(),
+            )
+            .await;
+        }
+    })
+    .abort_handle()
+}
+
+async fn remove_timed_out_rounds(
+    round_timeout: Duration,
+    initiated_round_state: Arc<Mutex<Option<GossipRoundState>>>,
+    accepted_round_states: Arc<Mutex<HashMap<Url, GossipRoundState>>>,
+) {
+    {
+        let mut initiated_state = initiated_round_state.lock().await;
+        match initiated_state.as_ref() {
+            Some(state) if state.started_at.elapsed() > round_timeout => {
+                tracing::warn!("Initiated round timed out: {:?}", state);
+                *initiated_state = None;
+            }
+            _ => (),
+        }
+    }
+
+    accepted_round_states.lock().await.retain(|_, state| {
+        match state.started_at.elapsed() > round_timeout {
+            true => {
+                tracing::warn!("Accepted round timed out: {:?}", state);
+                false
+            }
+            false => true,
+        }
```

Comment:
Is an if a little easier to read here?

```suggestion
        if state.started_at.elapsed() > round_timeout {
            tracing::warn!("Accepted round timed out: {:?}", state);
            false
        } else {
            true
        }
```